src/victory-legend/victory-legend.js

```diff
@@ -111,16 +111,18 @@ export default class VictoryLegend extends React.Component {
   getCalculatedProps() { // eslint-disable-line max-statements
     const { role } = this.constructor;
     const { data, orientation, theme } = this.props;
-    let { height, width } = this.props;
-
+    let { height, padding, width } = this.props;
     const legendTheme = theme && theme[role] ? theme[role] : {};
     const colorScale = this.getColorScale(legendTheme);
     const isHorizontal = orientation === "horizontal";
-    const padding = Helpers.getPadding(this.props);
     const symbolStyles = [];
     const labelStyles = [];
     let leftOffset = 0;
 
+    padding = Helpers.getPadding({ padding: padding || theme.padding });
```

Comment:
You'll need to remove padding from default props, and make this check respect padding = 0. Also, if you remove padding from default props, this should have a fallback right in the code like props.padding || theme.padding || 0; except with existence checking that respects zero.